src/plugins/intel_gpu/src/kernel_selector/kernels/cum_sum/cum_sum_kernel_ref.cpp

```diff
@@ -35,8 +35,30 @@ ParamsKey CumSumKernelRef::GetSupportedKey() const {
 
 JitConstants CumSumKernelRef::GetJitConstants(const cum_sum_params& params, DispatchData dispatchData) const {
     auto jits = CumSumKernelBase::GetJitConstants(params, dispatchData);
+    auto axis_idx = GetCumSumAxisIndex(params);
 
-    jits.AddConstant(MakeJitConstant("AXIS_LAYOUT_INDEX", GetCumSumAxisIndex(params)));
+    jits.AddConstant(MakeJitConstant("AXIS_LAYOUT_INDEX", axis_idx));
+
+    if (params.reverse) {
+        const auto& output = params.outputs[0];
+        if (output.is_dynamic()) {
+            const int rank = static_cast<int>(output.LogicalDims().size());
+            auto idx = rank - axis_idx - 1;
+            int shape_info_idx = idx;
+            if (idx >= 2) {
+                shape_info_idx += (6 - rank);
+            }
+
+            size_t num_of_dynamic_inputs = 0;
+            for (auto& output : params.inputs) {
+                if (output.is_dynamic()) {
```

Comment:
Variable naming is confusing. It is counting input right?

Reply:
Yes, it should be "input" instead of "output". Fixed.